xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -553,14 +555,17 @@ private void handleLdsResponseForConfigUpdate(DiscoveryResponse ldsResponse) {
     List<String> listenerNames = new ArrayList<>(ldsResponse.getResourcesCount());
     try {
       for (com.google.protobuf.Any res : ldsResponse.getResourcesList()) {
+        if (res.getTypeUrl().equals(ADS_TYPE_URL_LDS_V2)) {
+          res = res.toBuilder().setTypeUrl(ADS_TYPE_URL_LDS).build();
```

Comment:
Is this trick really gonna work?

Reply:
I think the answer is yes since that is the whole basis for this PR (or this set of PRs). Having a unit test to prove it works should not be difficult.

Reply:
Why not?

Reply:
I am suspecting `Any.unpack` would break. I really doubt `Any.unpack` can deserialize a v3 Listener bytes to a v2 Listener without screwing up the message. Do you have any doc reference doing that?

Reply:
@voidzcy the logic here is the opposite. We are trying to deserialize a v2 proto payload as v3. And the approach has been discussed and agreed to for this work (in the design doc as well as a pending gRFC). If the field numbers are maintained and proper handling is done for deprecated and new fields I don't see why it won't work.

Reply:
Ok, I see. They put placeholders for deprecated fields and arrange fields in the way that the message is ensured to be the same in the wire format as v2. Then this should work.

Reply:
resolving...